.buildkite/hooks/pre-command

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -euo pipefail
+
+function retry {
+    local retries=$1
+    shift
+
+    local count=0
+    until "$@"; do
+        exit=$?
+        wait=$((2 ** count))
+        count=$((count + 1))
+        if [ $count -lt "$retries" ]; then
+            >&2 echo "Retry $count/$retries exited $exit, retrying in $wait seconds..."
+            sleep $wait
+        else
+            >&2 echo "Retry $count/$retries exited $exit, no more retries left."
+            return $exit
+        fi
+    done
+    return 0
+}
+
+GCP_SERVICE_ACCOUNT_SECRET_PATH=secret/ci/elastic-elastic-package/gcp-service-account
+AWS_SERVICE_ACCOUNT_SECRET_PATH=kv/ci-shared/platform-ingest/aws_account_auth
+GITHUB_TOKEN_VAULT_PATH=kv/ci-shared/platform-ingest/github_token
+
+# Secrets must be redacted
+# https://buildkite.com/docs/pipelines/managing-log-output#redacted-environment-variables
+
+if [[ "$BUILDKITE_PIPELINE_SLUG" == "elastic-package" && "$BUILDKITE_STEP_KEY" == "integration-parallel-gcp" ]]; then
+    export ELASTIC_PACKAGE_GCP_PROJECT_SECRET=$(retry 5 vault read -field projectId ${GCP_SERVICE_ACCOUNT_SECRET_PATH})
+    export ELASTIC_PACKAGE_GCP_CREDENTIALS_SECRET=$(retry 5 vault read -field credentials ${GCP_SERVICE_ACCOUNT_SECRET_PATH})
+
+    # Environment variables required by the service deployer
+    export GOOGLE_CREDENTIALS=${ELASTIC_PACKAGE_GCP_CREDENTIALS_SECRET}
+    export GCP_PROJECT_ID=${ELASTIC_PACKAGE_GCP_PROJECT_SECRET}
```

Comment:
Should we set these [`redacted-vars`](https://buildkite.com/docs/agent/v3/configuration#redacted-vars) to hide these variables that doesn't match with the default patterns?

Reply:
As the environment variables ELASTIC_PACKAGE_GCP_PROJECT_SECRET and ELASTIC_PACKAGE_GCP_CREDENTIALS_SECRET match the pattern of redacted variables, if other environment variables are built from there, they are also redacted.

I did some tests in this PR https://github.com/elastic/elastic-package/pull/1133
and here it is the output https://buildkite.com/elastic/elastic-package/builds/207#01865643-9d81-42bc-8010-971526776081

It looks like buildkite takes into account the string contained by the environment variables, and any occurrence is redacted.

Reply:
Ah ok, nice then.